test/pytest/test_bridge.py

```diff
@@ -634,6 +634,73 @@ async def test_fsreplace1_error(transport: MockTransport, tmp_path: Path) -> Non
     })
 
 
+@pytest.mark.asyncio
+async def test_fsreplace1_size_hint(transport: MockTransport, tmp_path: Path) -> None:
+    myfile = tmp_path / 'myfile'
+
+    # create a file, no size hint
+    ch = await transport.check_open('fsreplace1', path=str(myfile))
+    assert list(tmp_path.iterdir()) == []  # no temp file should be created yet
+    transport.send_data(ch, b'content')
+    transport.send_done(ch)
+    await transport.assert_msg('', command='done', channel=ch)
+    await transport.check_close(ch)
+    assert list(tmp_path.iterdir()) == [myfile]  # no temp file left
+    assert myfile.read_bytes() == b'content'
+
+    # delete it again
+    ch = await transport.check_open('fsreplace1', path=str(myfile))
+    assert list(tmp_path.iterdir()) == [myfile]  # no temp file created
+    transport.send_done(ch)
+    await transport.assert_msg('', command='done', channel=ch)
+    await transport.check_close(ch)
+    assert list(tmp_path.iterdir()) == []  # empty again
+
+    # enospc in advance: 1PB ought to be enough for anybody...
+    ch = transport.send_open('fsreplace1', path=str(myfile), size=1 << 50)  # 1PB
+    err = await transport.assert_msg('', command='close', channel=ch, problem='internal-error')
+    message = get_str(err, 'message')
+    assert 'No space left on device' in message or 'File too large' in message
+    assert list(tmp_path.iterdir()) == []  # still empty
+    assert not myfile.exists()
+
+    # ensure cancellation is always a no-op, even with size specified
+    ch = await transport.check_open('fsreplace1', path=str(myfile), size=1 << 20)  # 1MB
+    assert len(list(tmp_path.iterdir())) == 1  # observe the temp file created immediately
+    await transport.check_close(ch)
+    assert list(tmp_path.iterdir()) == []  # no temp file left
+    assert not myfile.exists()
+
+    # but even a non-zero size, will result in the file being created without payload
+    ch = await transport.check_open('fsreplace1', path=str(myfile), size=0)
```

Comment:
`size=0` isn't a "non-zero size".; I suppose the comment wanted to say "even without sending data"?

Reply:
`s/non-//` thanks